src/pywrapper.cpp

```diff
@@ -102,26 +116,61 @@ bool PyWrapper::init()
     // communication channel for I/O Intr value exchange
     PyImport_AppendInittab("pydev", &PyInit_pydev);
 
+    // Py_SetProgramName(L"python3.7");
+    std::cerr << "Initalising python3" << std::endl;
     Py_Initialize();
+    std::cerr << "Initialsing threads" << std::endl;
     PyEval_InitThreads();
 
+    std::cerr << "Initialsing dics" << std::endl;
+
+#if PY_MAJOR_VERSION == 3
+
+    /*
+     * IN Python 3 the
+     */
+    auto m = PyImport_AddModule("__main__");
+    assert(m);
+    globDict = PyModule_GetDict(m);
+    assert(globDict);
+    locDict = PyDict_New();
+    assert(locDict);
+    /*     */
+#elif PY_MAJOR_VERSION == 2
     globDict = PyDict_New();
     locDict = PyDict_New();
     PyDict_SetItemString(globDict, "__builtins__", PyEval_GetBuiltins());
+#else /* PY_MAJOR_VERSION */
+#error "Unsupported python version"
+#endif /* PY_MAJOR_VERSION */
 
     mainThread = PyEval_SaveThread();
 
     // Make `pydev' module appear as built-in module
+    std::cerr << "Importing pydev" << std::endl;
+    exec("import pydev", true);
+    std::cerr << "Importing builtins" << std::endl;
+
+#if PY_MAJOR_VERSION == 3
+    exec("import builtins", true);
+    exec("builtins.pydev=pydev", true);
     exec("import pydev", true);
+#elif PY_MAJOR_VERSION == 2
     exec("import __builtin__", true);
     exec("__builtin__.pydev=pydev", true);
+#else
+#error "Unsupported python version"
+#endif
+
+    std::cerr << "python imported" << std::endl;
 
     return true;
 }
 
 void PyWrapper::shutdown()
 {
-    PyEval_AcquireLock();
+    // PyEval_AcquireLock();
```

Comment:
I agree with use of PyEval_AcquireThread(), let's simply remove the PyEval_AcquireLock() without the comment.

Reply:
Ok